Docs: Headless CMS (#1831)
vtex/faststore: apps/site/pages/docs/headless-cms-integration/2-setting-up-vtex-headless-cms.mdx

---
title: '2. Setting up the VTEX Headless CMS in your FastStore project'
sidebar_label: 'headless-cms-integration'
---

import { Callout } from 'nextra-theme-docs'

<header>
# Part 2: Setting up the VTEX Headless CMS in your FastStore project
</header>

Now that you are familiar with the core concepts of the [**VTEX Headless CMS**](/docs/headless-cms-overview/overview-headless-cms) and have [set it up in your VTEX account](/docs/headless-cms-integration/1-configuring-the-vtex-account), let's get back to our **FastStore** project to define our own **Content Types** and **Sections**.

By the end of this part of this tutorial, you will be able to see the definition of a Content Type available for use at the VTEX Headless CMS app. 

---

## Step by step

### Step 1 - Creating the CMS folder

1. Open the terminal and change to the root directory of your FastStore project.

> 2. Create a new folder named `cms` at the root of your FastStore project:

Author: ```suggestion
2. Create a new folder named `cms` at the root of your FastStore project. Inside `cms` create another folder for your project:
```

    ```bash
    mkdir cms
    ```

3. Inside the `cms` folder, create a folder for your project:

    ```bash

>     mkdir cms/faststore

Author: This command doesn't work for Windows users, the correct one for them is: mkdir cms\faststore

---
title: '2. Setting up the VTEX Headless CMS in your FastStore project'
sidebar_label: 'headless-cms-integration'
---

import { Callout } from 'nextra-theme-docs'

<header>
# Part 2: Setting up the VTEX Headless CMS in your FastStore project
</header>

> Now that you are familiar with the core concepts of the [**VTEX Headless CMS**](/docs/headless-cms-overview/overview-headless-cms) and have [set it up in your VTEX account]/docs/headless-cms-integration/1-configuring-the-vtex-account), let's get back to our **FastStore** project to define our own **Content Types** and **Sections**.

Author: ```suggestion
Now that you are familiar with the core concepts of the [**VTEX Headless CMS**](/docs/headless-cms-overview/overview-headless-cms) and have [set it up in your VTEX account](/docs/headless-cms-integration/1-configuring-the-vtex-account), let's get back to our **FastStore** project to define our own **Content Types** and **Sections**.
```

By the end of this part of this tutorial, you will be able to see the definition of a Content Type available for use at the VTEX Headless CMS app. 

---

## Step by step

### Step 1 - Creating the CMS folder

1. Open the terminal and change to the root directory of your FastStore project.
2. Create a new folder named `cms` at the root of your FastStore project:

    ```bash
    mkdir cms
    ```

3. Inside the `cms` folder, create a folder for your project:

    ```bash

>     cd cms

Author: ```suggestion
```

>     mkdir faststore

Author: ```suggestion
    mkdir cms/faststore
```

| File name                | Description                                                                                                                                                    |
| ------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `content-types.json`     |  An array of JSON objects that describes the **Content Types** available for customization at the VTEX Headless CMS app.                                       |
| `sections.json`          | An array of JSON objects that describes the content structure of the frontend **Section** components available for customization at the VTEX Headless CMS app. |
|  `translation-keys.json` | an array of JSON objects that defines the translation keys of the Sections descriptions.                                                                       |

```bash
    touch cms/faststore/content-types.json cms/faststore/sections.json cms/faststore/translation-keys.json
```

5. Update the `sections.json` file with an empty array:

```bash
   echo "[]" > cms/faststore/sections.json
```

> 6. Update the `translation-keys.json` file with an empty object. *Remember to replace the values between curly brackets according to your account name.*:

Author: ```suggestion
6. Update the `translation-keys.json` file with an empty object. :
```

```bash
   echo "{}" > cms/faststore/translation-keys.json
```

7. Open the `content-types.json` file in any code editor of your choice and add the following code:

```json title="cms/faststore/content-types.json"

>     [

Reviewer: Update this example.

>     [
>       {
>         "id": "home",
>         "name": "Home Page",
>         "configurationSchemaSets": []
>       },
>       {
>         "id": "institutionalPage",
>         "name": "Institutional page",
>         "configurationSchemaSets": [
>           {
>             "name": "SEO",
>             "configurations": [
>               {
>                 "name": "siteMetadataWithSlug",
>                 "schema": {
>                   "title": "Site Metadata",
>                   "description": "Configure global site metadata",
>                   "type": "object",
>                   "widget": {
>                     "ui:ObjectFieldTemplate": "GoogleSeoPreview"
>                   },
>                   "properties": {
>                     "title": {
>                       "title": "Default page title",
>                       "description": "Display this title when no other tile is available",
>                       "type": "string",
>                       "default": "Store Theme | VTEX SFJ"
>                     },
>                     "description": {
>                       "title": "Meta tag description",
>                       "type": "string",
>                       "default": "A beautifully designed site for general VTEX stores"
>                     },
>                     "titleTemplate": {
>                       "title": "Title template to be used in category/product pages",
>                       "type": "string",
>                       "default": "%s | Store Theme"
>                     },
>                     "slug": {
>                       "title": "URL Slug",
>                       "type": "string",
>                       "default": "/landing-page-url"
>                     }
>                   }
>                 }
>               }
>             ]
>           }
>         ]
>       }
>     ]

Reviewer: ```suggestion
[
  {
    "id": "customPage",
    "name": "Custom Page",
    "configurationSchemaSets": [
      {
        "name": "Settings",
        "configurations": [
          {
            "name": "seo",
            "schema": {
              "title": "SEO",
              "description": "Search Engine Optimization options",
              "type": "object",
              "widget": {
                "ui:ObjectFieldTemplate": "GoogleSeoPreview"
              },
              "required": [
                "slug",
                "title",
                "description"
              ],
              "properties": {
                "slug": {
                  "title": "Path",
                  "type": "string",
                  "default": "/"
                },
                "title": {
                  "title": "Default page title",
                  "description": "Display this title when no other title is available",
                  "type": "string",
                  "default": "FastStore Starter"
                },
                "description": {
                  "title": "Meta tag description",
                  "type": "string",
                  "default": "A beautifully designed store"
                },
                "canonical": {
                  "title": "Canonical url for the page",
                  "type": "string"
                }
              }
            }
          }
        ]
      }
    ]
  }
]
```

```
<Callout type="info" emoji="ℹ️">
  Don't worry about the structure of this file for now, as we'll learn more about it later in this tutorial. However, notice that we have defined two different Content Types: the **Home Page** and the **Institutional Page**.
</Callout>

8. Save your changes in the `content-types.json` file.

### Step 2 - Syncing your changes

Let's now sync our changes with the VTEX Headless CMS app and see what happens.

1. Open the terminal and log in to your VTEX account.

2. Create a new development workspace by running the following command. *Remember to replace the values between curly brackets according to your scenario*:

```bash
   vtex use {workspace}
```

<Callout type="warning" emoji="⚠️">
  Use a development workspace to try your definitions of Sections and Content Types before syncing them with the VTEX `master` workspace.
</Callout>

3. Change to the root directory of your FastStore project.

4. Sync your changes in the `cms` folder with the VTEX Headless CMS app:

```bash

>     vtex cms sync faststore

Reviewer: ```suggestion
    yarn faststore cms-sync
```

```

<Callout type="warning" emoji="⚠️">
   If you have the `@vtex/cli-plugin-cms@1.0.7` installed you can use the `{builderName}` argument to syncronize a specific builder: `vtex cms:sync {builderName}`.
</Callout>

Once your changes are synced with the VTEX Headless CMS, the terminal will show the following message.

```bash
  CMS synced successfully...
```

### Step 3 - Checking your changes

1. Access the VTEX Admin using the workspace you previously created (e.g., `https://{workspace}--{account}.myvtex.com/admin`).
2. Go to **Storefront > Headless CMS**.

> 3. Select your Faststore project.
> 4. Click on **Create New**.

Reviewer: ```suggestion
3. Click on **Create New**.
```

> You should now see the Content Type we created in the previous step available for use at the VTEX Headless CMS app. However, no sections or translation keys will be available yet. We'll learn more about this in the following part of this tutorial.

Reviewer: Add image example?
